common/dynamicconfig/constants.go

```diff
@@ -24,1049 +24,2292 @@
+	)
+	ScannerPersistenceMaxQPS = NewIntGlobalSetting(
+		"worker.scannerPersistenceMaxQPS",
+		100,
+		`ScannerPersistenceMaxQPS is the maximum rate of persistence calls from worker.Scanner`,
+	)
+	ExecutionScannerPerHostQPS = NewIntGlobalSetting(
+		"worker.executionScannerPerHostQPS",
+		10,
+		`ExecutionScannerPerHostQPS is the maximum rate of calls per host from executions.Scanner`,
+	)
+	ExecutionScannerPerShardQPS = NewIntGlobalSetting(
+		"worker.executionScannerPerShardQPS",
+		1,
+		`ExecutionScannerPerShardQPS is the maximum rate of calls per shard from executions.Scanner`,
+	)
+	ExecutionDataDurationBuffer = NewDurationGlobalSetting(
+		"worker.executionDataDurationBuffer",
+		time.Hour*24*90,
+		`ExecutionDataDurationBuffer is the data TTL duration buffer of execution data`,
+	)
+	ExecutionScannerWorkerCount = NewIntGlobalSetting(
+		"worker.executionScannerWorkerCount",
+		8,
+		`ExecutionScannerWorkerCount is the execution scavenger worker count`,
+	)
+	ExecutionScannerHistoryEventIdValidator = NewBoolGlobalSetting(
+		"worker.executionEnableHistoryEventIdValidator",
+		true,
+		`ExecutionScannerHistoryEventIdValidator is the flag to enable history event id validator`,
+	)
+	TaskQueueScannerEnabled = NewBoolGlobalSetting(
+		"worker.taskQueueScannerEnabled",
+		true,
+		`TaskQueueScannerEnabled indicates if task queue scanner should be started as part of worker.Scanner`,
+	)
+	BuildIdScavengerEnabled = NewBoolGlobalSetting(
+		"worker.buildIdScavengerEnabled",
+		false,
+		`BuildIdScavengerEnabled indicates if the build id scavenger should be started as part of worker.Scanner`,
+	)
+	HistoryScannerEnabled = NewBoolGlobalSetting(
+		"worker.historyScannerEnabled",
+		true,
+		`HistoryScannerEnabled indicates if history scanner should be started as part of worker.Scanner`,
+	)
+	ExecutionsScannerEnabled = NewBoolGlobalSetting(
+		"worker.executionsScannerEnabled",
+		false,
+		`ExecutionsScannerEnabled indicates if executions scanner should be started as part of worker.Scanner`,
+	)
+	HistoryScannerDataMinAge = NewDurationGlobalSetting(
+		"worker.historyScannerDataMinAge",
+		60*24*time.Hour,
+		`HistoryScannerDataMinAge indicates the history scanner cleanup minimum age.`,
+	)
+	HistoryScannerVerifyRetention = NewBoolGlobalSetting(
+		"worker.historyScannerVerifyRetention",
+		true,
+		`HistoryScannerVerifyRetention indicates the history scanner verify data retention.
+If the service configures with archival feature enabled, update worker.historyScannerVerifyRetention to be double of the data retention.`,
+	)
+	EnableBatcherGlobal = NewBoolGlobalSetting(
+		"worker.enableBatcher",
+		true,
+		`EnableBatcher decides whether to start old (system namespace) batcher in our worker`,
+	)
+	EnableBatcherNamespace = NewBoolNamespaceSetting(
+		"worker.enableNamespaceBatcher",
+		true,
+		`EnableBatcher decides whether to start new (per-namespace) batcher in our worker`,
+	)
```

Comment:
This was a bug, or at least a very confusing situation: the same key was being used as a global setting and a per-namespace setting. I split this into two separate keys.

Reply:
Yeah, it does read a little better. I think I can do that with a bit of sed